CMakeLists.txt

```diff
@@ -67,6 +67,8 @@ if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
     if(BUILD_TESTS)
         enable_testing()
         include(CTest)
+        add_definitions(-DPROTECTED=public)
+        add_definitions(-DPRIVATE=public)
```

Comment:
For the sake of symmetry, can this live under `test/test_unit/CMakeList.txt` instead of here in the root CMakeLists?